Example/GiniVision/ComponentAPICoordinator.swift

```diff
@@ -168,151 +206,323 @@ final class ComponentAPICoordinator: NSObject, Coordinator {
+                              to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
         
-        if fromViewController is ComponentAPIReviewViewController &&
-            viewController is ComponentAPICameraViewController {
+        if fromVC is ReviewViewController && operation == .pop {
             reviewScreen = nil
-            documentService.cancelAnalysis()
+            if let document = documentRequests.first?.document {
+                documentService?.delete(document)
+            }
+            documentRequests.removeAll()
         }
         
-        if fromViewController is ComponentAPIAnalysisViewController &&
-            viewController is ComponentAPIReviewViewController {
+        if fromVC is AnalysisViewController && operation == .pop {
+            // Going directly from the analysis to the camera means that
+            // the document is not an image and should be removed
+            if toVC is CameraViewController {
+                documentRequests.removeAll()
+            }
+            
             analysisScreen = nil
-            documentService.cancelAnalysis()
+            documentService?.cancelAnalysis()
         }
         
-        if let resultsScreen = fromViewController as? ResultTableViewController {
-            documentService.sendFeedback(forDocument: resultsScreen.document)
+        if let resultsScreen = fromVC as? ResultTableViewController {
+            documentService?.sendFeedback(with: resultsScreen.result)
         }
+        
+        if let cameraViewController = toVC as? CameraViewController,
+            fromVC is MultipageReviewViewController {
+            cameraViewController
+                .replaceCapturedStackImages(with: documentRequests.compactMap { $0.document.previewImage })
+        }
+        
+        return nil
     }
 }
 
-// MARK: ComponentAPICameraScreenDelegate
+// MARK: - CameraViewControllerDelegate
 
-extension ComponentAPICoordinator: ComponentAPICameraViewControllerDelegate {
-    func componentAPICamera(viewController: UIViewController, didPickDocument document: GiniVisionDocument) {
-        if document.isReviewable {
-            showReviewScreen(withDocument: document)
-        } else {
-            showAnalysisScreen(withDocument: document)
+extension ComponentAPICoordinator: CameraViewControllerDelegate {
+    func camera(_ viewController: CameraViewController, didCapture document: GiniVisionDocument) {
+        validate([document]) { result in
+            switch result {
+            case .success(let documentRequests):
+                self.documentRequests.append(contentsOf: documentRequests)
+                self.upload(documentRequests: documentRequests)
+                
+                if self.giniConfiguration.multipageEnabled, self.documentRequests.count > 1 {
+                    if let imageDocument = document as? GiniImageDocument {
+                        viewController.animateToControlsView(imageDocument: imageDocument)
```

Comment:
Another closure two-humped camel of doom.

Reply:
In this case I don't agree with you. Extracting this into a different method would make much more complicated.